doc/tools/pkcs11-tool.1.xml

```diff
@@ -146,9 +146,9 @@
 					<term>
 						<option>--key-type</option> <replaceable>specification</replaceable>
 					</term>
-					<listitem><para>Specify the type and length (bytes if symmetric) of the key to create,
+					<listitem><para>Specify the type and (optional) flavour of the key to create,
```

Comment:
I think this is still not completely accurate as I think the `GENERIC` secret always need the length if I remember well. I would probably keep the length naming as "length" is usually more understandable for the users than the flavour. I think the original note `(bytes if symmetric)` was meant as "bits for asymmetric and bytes for symmetric), which got lost by this change. Do we have some some other key type where the length is optional than AES and for unwrapping? If not, I would keep it as specific as possible:
```suggestion
					<listitem><para>Specify the colon separated type and length (optional for AES unwrapping) of the key to create,
```

Reply:
If we strictly stick to the PKCS#11 spec, the presence of CKA_VALUE_LEN in the template is:
- for unwrapping **never** _compulsory_, **sometimes** _forbidden_,
- for key creation (C_CreateObject) **always** _forbidden_,
- for key generation **always** _compulsory_.

Also the reason I changed "length" to "flavour" is because in many cases the "thing" that follows the key type is not a length, but an elliptic curve identifier or another kind of parameter.
I agree that we are loosing the byte/bit length differentiation, but, now that we're on this road, it feels weird to mention an "AES:16" key. The more traditional 128 bit-wise length could be processed in the program to send the expected byte-wise value to the P11 module.

I feel like this CLI parameter fills too big of a role and also needs to work with plugins that do not strictly follow the same conventions so I find it very difficult to give it an exhaustive, concise description.
IMO the correct thing to do would be to separate it, but it's hardly ideal since it would break retrocompatibility.
For example have ```key type``` alone, a ```key length``` that would result in a CKA_VALUE_LEN for symmetric key objects and CKA_MODULUS_BITS for RSA keys and a ```key flavour``` for everything else. Or merge ```key length``` and ```key flavour```.

Would something of this extent be more acceptable ? It feels awkward but at least its somehow complete...
> Specify the type and (not always compulsory) flavour (byte-wise symmetric key length, bit-wise asymmetric key length, elliptic curve identifier, etc.) of the key to create, for example RSA:1024, EC:prime256v1, GOSTR3410-2012-256:B, DES:8, DES3:24, AES:16, AES: or GENERIC:64.

Reply:
Yes, I think this should be enough. Thank you for having a look into that!

Once we touch it, it might make sense to use some better example for RSA, as people tend to copy from manual pages and use that. Having 2k in examples might work better.

Reply:
> key flavour for everything else. 
Since pkcs11-tool is for PKCS11 we should stick to PKCS11 terms.  As you pointed out  CKA_*_LEN is in bytes and CKA_*_BITS is in bits.

All EC mechanisms use CKA_EC_PARAMS which has 4 choices: ECParameters, OID curveNames, implicitlyCA and  curveName as PrintableString.   "The use of oId or curveName is recommended over the choice ecParameters. The choice implicitlyCA must not be used in Cryptoki." 

standardized as OID curve names.  was added as some curves names where bases on IETF experimental curve names,  and implemented in software and hardware but were later defined with standardized as OID curveNames. 
See https://github.com/OpenSC/OpenSC/issues/3000  and https://github.com/OpenSC/OpenSC/pull/3090  
(which is not complete), but has several changes for pkcs11-tool to use standardized OID curve names and address the use of standardized as OID curveNames. 

I have not looked at all the PKCS11 defined mechanisms but DES, AES, RSA, EC and maybe GOST are the common ones supported by OpenSC.  So "flavor" comes down to OID curveNames.